### ISSUE TITLE
Catched View exception still messes up page output

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2#13058]
Migrated from https://github.com/yiisoft/yii2-swiftmailer/issues/38

### What steps will reproduce the problem?

1) Create web controller action:

```php
    public function actionRender()
    {
        try {
            return $this->render('exception');
        } catch (\Exception $e) {
            return $this->render('index');
        }
    }
```

2) Create view file `exception.php` which throws some exception inside:

```php
<h1>File Begin</h1>
<?php throw new Exception('test'); ?>
<h1>File End</h1>
```

### What is the expected result?

Fallback view rendered correctly

### What do you get instead?

Line `<h1>File Begin</h1>` from the view, which causes an exception appears at the top of the render result.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 5.6
| Operating system | CentOS

**Comment by klimov-paul** [comment]
I am unsure if this actually can be fixed.
Obvious solution will be wrap `View::render()` internal code with `try..catch` block and call `ob_clean()` on exception.
However, output buffering may go deeper during single file rendering, in case such widgets as `ActiveForm`, `Spaceless` or `FragmentCache` are used.

We can clear all output buffering like it is done at `ErrorHandler::clearOutput()`:
https://github.com/yiisoft/yii2/blob/master/framework/base/ErrorHandler.php#L295-L303
But this also may break application logic.

**Comment by klimov-paul** [comment]
I have found the way to fix this, see #14239
